### PR TITLE
ITK: Added patron reg. redirect if logged in

### PR DESCRIPTION
**Pull request by cableman**
#### Link to issue

https://jira.itkdev.dk/browse/CMN-647

#### Description

User accessing `/user/registration` will be redirect to `\user\me` if they are logged in.

#### Screenshot of the result

N/A

#### Checklist

- [x] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [x] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [x] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

#### Additional comments or questions

N/A

### CODE REVIEW

File: web/modules/custom/dpl_patron_reg/src/Controller/DplPatronRegController.php [hunk 1]
```diff
@@ -61,37 +61,48 @@ public static function create(ContainerInterface $container): DplPatronRegContro
...
+        if (!$this->config('openid_connect.settings.' . $client_id)
+          ->get('enabled')) {
+          continue;
+        }
+
+        $url = Url::fromRoute('dpl_patron_reg.auth', ['client_name' => $client_id], ['absolute' => TRUE]);
+        $link = Link::fromTextAndUrl($this->t('Log in with @client_title', [
+          '@client_title' => $client['label'],
+        ]), $url);
+        $logins[$client_id] = $link->toRenderable();
```
Comment (spaceo) [inline web/modules/custom/dpl_patron_reg/src/Controller/DplPatronRegController.php:87]: I find it confusing and time consuming to figure out when what is returned and when `$output` is used and when not.
I suggest this formatting instead:
```php
  public function informationPage(): mixed {
    if (!$this->currentUser()->isAnonymous()) {
      /** @var \Drupal\Core\GeneratedUrl $url */
      $url = Url::fromRoute('dpl_patron_page.profile')->toString(TRUE);
      return new TrustedRedirectResponse($url->getGeneratedUrl());
    }

    $config = $this->patronRegSettings->loadConfig();
    $logins = [];
    // Loop over all open id connect definitions and build login links for
    // each one.
    $definitions = $this->pluginManager->getDefinitions();
    foreach ($definitions as $client_id => $client) {
      if (!$this->config('openid_connect.settings.' . $client_id)
        ->get('enabled')) {
        continue;
      }

      $url = Url::fromRoute('dpl_patron_reg.auth', ['client_name' => $client_id], ['absolute' => TRUE]);
      $link = Link::fromTextAndUrl($this->t('Log in with @client_title', [
        '@client_title' => $client['label'],
      ]), $url);
      $logins[$client_id] = $link->toRenderable();
    }

    return [
      'info' => [
        '#type' => 'processed_text',
        '#text' => $config->get('information')['value'] ?? 'Please fill out the information page in the administration',
        '#format' => $config->get('information')['format'] ?? 'plain_text',
      ],
      'logins' => $logins,
    ];

  }

**Review by spaceo — APPROVED** [review]
Looks good.
Please look at my suggestion for restructuring method.

File: web/modules/custom/dpl_patron_reg/src/Controller/DplPatronRegController.php (hunk 1 above)
Comment (cableman) [reply web/modules/custom/dpl_patron_reg/src/Controller/DplPatronRegController.php]: Agreed :-) and fixed

**Comment by cableman** [comment]
Refactored as suggested and ready for re-review

**Review by sinejespersen — APPROVED** [review]

**Review by spaceo — APPROVED** [review]
LGTM 👍

**Pull request merged**